detection_rules/custom_schemas.py

```diff
@@ -47,11 +47,9 @@ def resolve_schema_path(path: str) -> Path:
     return path_obj if path_obj.is_absolute() else RULES_CONFIG.stack_schema_map_file.parent.joinpath(path)
 
 
-def update_data(index: str, field: str, data: dict) -> dict:
+def update_data(index: str, field: str, data: dict, known_type: str = None) -> dict:
```

Comment:
would `field_type` instead of `known_type` be more descriptive here and in the `update_auto_generated_schema` ?

Reply:
Yes, completely agree. known_type does not communicate what it will be used for, will update :+1: